lib/backend/dynamo/dynamodbbk.go

```diff
@@ -118,6 +132,10 @@ func (cfg *Config) CheckAndSetDefaults() error {
 		cfg.RetryPeriod = defaults.HighResPollingPeriod
 	}
 
+	if cfg.BillingMode == billingModeOnDemand {
+		cfg.EnableAutoScaling = false
+	}
```

Comment:
This breaks existing config files for existing tables that are setting/changing their autoscaling settings.

Reply:
Would it make sense to only disable auto scaling if 
- the table does not already exist, is being created and the config is in on-demand mode 
- the table exists and is already in on demand mode 

I think that would stop it being a breaking change